app/Insightly/InsightlyContactLink.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Insightly;
+
+use App\Domain\Contacts\Contact;
+use Illuminate\Support\Arr;
+
+final class InsightlyContactLink implements ContactLink
+{
+    public function __construct(private readonly InsightlyClient $insightlyClient)
+    {
+    }
+
+    public function link(Contact $contact): int
```

Comment:
For me, it is not so clear what this method does. Maybe it is better for readability to just inline this inside the handlers?

Reply:
Maybe a better name is something like `getUniqueContact` which can become part of `ContactResource`?

Reply:
Or even better modify `ContactResource:create` to enforce uniqueness

Reply:
Thinking about it, it feels more like a `findOrCreateByEmail()` function. The hard thing is determining where we should guard the domain rule that Contacts in Insightly stay unique on email address.